File: LeetCode/Medium/1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        ret = 0
        prev = colors[0]
        prev_time = neededTime[0]

        for i in range(1, len(colors)):
            if colors[i] == prev:
                ret += min(prev_time, neededTime[i])
                prev_time = max(prev_time, neededTime[i])
            else:
                prev = colors[i]
                prev_time = neededTime[i]
                
        return ret

File: LeetCode/Medium/1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int ret = 0;
        char prev = colors[0];
        int prevTime = neededTime[0];
        
        for (int i = 1; i < neededTime.size(); ++i){
            if (prev == colors[i]){
                ret += min(prevTime, neededTime[i]);
                prevTime = max(prevTime, neededTime[i]);
            } else {
                prev = colors[i];
                prevTime = neededTime[i];
            }
        }
        
        return ret;
    }
};